### PR TITLE
Add Supply Depot plugin

### PR DESCRIPTION
**Pull request by Naughty-Klaus**
A plugin that allows you too coordinate supply runs with other players.

- Single-item orders only currently
- Can cancel orders
- Item search function
- Uses custom-built API, only identifying information collected is display name. Canceling orders deletes this record.

### CODE REVIEW

**Comment by raiyni** [comment]
Don't close and open new prs. You can push updated commit hashes to the same branch.

**Comment by raiyni** [comment]
Don't close your PR. Just push the commit.

**Comment by Nightfirecat** [comment]
What is these ad.pdn files ([1](https://github.com/Naughty-Klaus/supply-depot/blob/80a76e00394a8bef64b53a6de514ba028f33eec9/src/main/resources/ads/ad.pdn) and [2](https://github.com/Naughty-Klaus/supply-depot/blob/80a76e00394a8bef64b53a6de514ba028f33eec9/src/main/java/net/naughtyklaus/depot/ad.pdn)), and why are you distributing them?

File: plugins/supply-depot [hunk 1]
```diff
@@ -0,0 +1,2 @@
+repository=https://github.com/Naughty-Klaus/supply-depot.git
+commit=80a76e00394a8bef64b53a6de514ba028f33eec9
```
Comment (Nightfirecat) [inline plugins/supply-depot:2]: The plugin needs a `warning` as it makes outbound connections with the default configuration.
```suggestion
commit=80a76e00394a8bef64b53a6de514ba028f33eec9
warning=This plugin submits your username and IP address and in game stats to a 3rd party server not controlled or verified by the RuneLite Developers.
```

**Comment by Naughty-Klaus** [comment]
@Nightfirecat I forgot to add them to gitignore, they're source images that am no longer using. I'll remove them.

File: plugins/supply-depot [hunk 2]
```diff
@@ -0,0 +1,3 @@
+repository=https://github.com/Naughty-Klaus/supply-depot.git
+commit=ee716a1f69e1c19df30ea633cefcbf3177890ae8
+warning=This plugin submits your username and IP address and in game stats to a 3rd party server not controlled or verified by the RuneLite Developers.
```
Comment (Felanbird) [inline plugins/supply-depot:3]:
```suggestion
warning=This plugin submits your username, IP address, and in game stats to a 3rd party server not controlled or verified by the RuneLite Developers.
```
nitpick from NFC's suggestion

Comment (Nightfirecat) [reply plugins/supply-depot:3]: Actually ingame stats aren't transmitted--I apparently copied the wrong warning from another plugin to use as an example here. Instead, I'll commit this:
```suggestion
warning=This plugin submits your username and IP address to a 3rd party server not controlled or verified by the RuneLite Developers.
```

Comment (Naughty-Klaus) [reply plugins/supply-depot]: Alright thank you, I also agree with Felanbird. Grammar really is important.

**Pull request merged**